### PR TITLE
Added 'Alert on HTTP Response Code Errors.js'

### PR DESCRIPTION
**Pull request by psiinon**
Depends on https://github.com/zaproxy/zaproxy/pull/3469 for registering the scanner id.

### CODE REVIEW

File: httpsender/Alert on HTTP Response Code Errors.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+pluginid = 100000	// https://github.com/zaproxy/zaproxy/blob/develop/src/doc/scanners.md
+
+function sendingRequest(msg, initiator, helper) {
+	// Nothing to do
+}
+
+function responseReceived(msg, initiator, helper) {
+	var extensionAlert = org.parosproxy.paros.control.Control.getSingleton().getExtensionLoader().getExtension(
+		org.zaproxy.zap.extension.alert.ExtensionAlert.class)
```
Comment (thc202) [inline httpsender/Alert on HTTP Response Code Errors.js]: I'm getting an error when running with Rhino, `EvaluatorException: missing name after . operator (<Unknown source>#17)`.
It works fine with both engines when used the extension name.

File: httpsender/Alert on HTTP Response Code Errors.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+				// Server error
+				risk = 1	// Low
+				title = "A Server Error response code was returned by the server"
+			}
+			// CONFIDENCE_HIGH = 3 (we can be pretty sure we're right)
+			var alert = new org.parosproxy.paros.core.scanner.Alert(pluginid, risk, 3, title)
+			var ref = msg.getHistoryRef()
+			if (ref == null) {
+				ref = new org.parosproxy.paros.model.HistoryReference(
+					org.parosproxy.paros.model.Model.getSingleton().getSession(), initiator, msg)
```
Comment (thc202) [inline httpsender/Alert on HTTP Response Code Errors.js]: Should it use other value for the history type? The ID of the initiators do not match the history types (should map the initiator ID to appropriate history type?).

**Comment by psiinon** [comment]
Updated to fix those issues

**Comment by thc202** [comment]
Thanks!

**Pull request merged**